api/handler/v1/runtime.go

```diff
@@ -656,12 +656,30 @@ func (sv *RuntimeServiceServer) RegisterSecret(ctx context.Context, req *pb.Regi
 		return nil, status.Errorf(codes.NotFound, "%s: project %s not found", err.Error(), req.GetProjectName())
 	}
 
-	secretRepo := sv.secretRepoFactory.New(projSpec)
-	if err := secretRepo.Save(ctx, models.ProjectSecretItem{
-		Name:  req.GetSecretName(),
-		Value: string(base64Decoded),
-	}); err != nil {
-		return nil, status.Errorf(codes.Internal, "%s: failed to save secret %s", err.Error(), req.GetSecretName())
+	namespaceSpec := models.NamespaceSpec{}
+	if req.GetNamespaceName() != "" {
+		namespaceRepo := sv.namespaceRepoFactory.New(projSpec)
+		namespaceSpec, err = namespaceRepo.GetByName(ctx, req.GetNamespaceName())
+		if err != nil {
+			return nil, status.Errorf(codes.NotFound, "%s: namespace %s not found", err.Error(), req.GetNamespaceName())
+		}
+	}
+
+	secretRepo := sv.secretRepoFactory.New(projSpec, namespaceSpec)
+	if req.UpdateOnly {
+		if err := secretRepo.Update(ctx, models.ProjectSecretItem{
+			Name:  req.GetSecretName(),
+			Value: string(base64Decoded),
+		}); err != nil {
+			return nil, status.Errorf(codes.Internal, "%s: failed to update secret %s", err.Error(), req.GetSecretName())
+		}
+	} else {
+		if err := secretRepo.Save(ctx, models.ProjectSecretItem{
```

Comment:
wouldn't the current implementation of `Save` will fail this if the secret is already registered? I was thinking the behavior should be 
- if the `set` is called and it doesn't exist, it will create it
- if the `set` is called and it does exist, it will update it
- if the `set` is called with `--update-only` flag and it doesn't exist, it will fail
- if the `set` is called with `--update-only` flag and it does exist, it will update

Reply:
The current behaviour will update an existing secret at time of create.
If this is not going to be a problem, then I can keep the current behaviour.

Reply:
Sorry didn't get what you said, can you rephrase it, please? Is the current behavior match with those 4 points?

Reply:
Existing behaviour (without change on this PR), save will create a new secret, if it exists, it will update the secret.
It might be a problem when the update of existing secret is unintentional at the time of create.

If we encounter this problem very less(unintentional update) then we can retain the existing behaviour.
Otherwise it will be better to be intentional about create or update of secrets.